app/front-end/src/features/editor/components/fileTreeView/fileTreeItem/fileTreeItem.tsx

```diff
@@ -174,7 +182,11 @@ export const FileTreeItem = React.forwardRef(function CustomTreeItem(
             onClick: (event) => {
               if (!blocked) {
                 if (getContentProps().onClick) getContentProps().onClick(event);
-                handleClick(item.id, item.label, item.fileType);
+                if (unsaved) {
+                  if (item.fileType !== FileTypes.FOLDER) setIsConfirmDialogOpen(true);
+                } else {
+                  handleClick(item.id, item.label, item.fileType);
+                }
               }
             },
 
```

Comment:
Also when using context menu actions such as `rename` and `export`, it should also prompt with unsaved data will be lost.

Reply:
Done